docs/sphinx/dev_guide/ci_tasks.rst

```diff
@@ -157,6 +157,128 @@ annotate the job for this. For example:
+that it can be accessed by the gitlab CI system. Since the intel compiler does
+not do releases in the typical sense (they simply update their repo *every night*), 
+it may become necessary to periodically build a new version of the compiler to 
+ensure that we are using the most up-to-date version available. The steps for 
+building, installing, and running are shown here.
+
+Building the Compiler
+^^^^^^^^^^^^^^^^^^^^^
+
+.. important:: Because intel updates their compiler repo daily, there is a nonzero possibility that the head of the sycl branch will fail to build. 
+  In the event that it does not build, try checking out a different commit. On the intel/llvm GitHub page, one can see which of their 
+  commits builds by checking the status badge next to each commit. Look for a commit that passes. 
+
+#. Load the version of GCC that you want to use. in this case, we are using LC's gcc/10.3.1-magic installation. 
+
+  .. code-block:: bash
+
+    module load gcc/10.3.1-magic
+
+#. Clone the "sycl" branch of intel's llvm compiler fork 
+
+  .. code-block:: bash
+
+    git clone https://github.com/intel/llvm -b sycl
+
+#. cd into that folder 
+
+  .. code-block:: bash
+    
+    cd llvm
+
+  #. In the event that the head of the sycl branch does not build, run `git checkout <git sha>` to checkout a version that does build.
+
+#. Build the compiler. Note that in this example, we are using rocm5.7.1, but one can change the version they wish to use simply by changing the paths in the configure step.
+
+  #. Configure
+
+    .. code-block:: bash 
+
+      srun -n1 /usr/bin/python3 buildbot/configure.py --hip -o buildrocm5.7.1 \
+      --cmake-gen "Unix Makefiles" \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_ROCM_DIR=/opt/rocm-5.7.1 \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_ROCM_INCLUDE_DIR=/opt/rocm-5.7.1/include \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_ROCM_LIB_DIR=/opt/rocm-5.7.1/lib \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_INCLUDE_DIR=/opt/rocm-5.7.1/include \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_HSA_INCLUDE_DIR=/opt/rocm-5.7.1/hsa/include/hsa \
+      --cmake-opt=-DSYCL_BUILD_PI_HIP_LIB_DIR=/opt/rocm-5.7.1/lib \
+      --cmake-opt=-DUR_HIP_ROCM_DIR=/opt/rocm-5.7.1 \
+      --cmake-opt=-DUR_HIP_INCLUDE_DIR=/opt/rocm-5.7.1/include \
+      --cmake-opt=-DUR_HIP_HSA_INCLUDE_DIR=/opt/rocm-5.7.1/hsa/include/hsa \
+      --cmake-opt=-DUR_HIP_LIB_DIR=/opt/rocm-5.7.1/lib
+  
+  #. Build
+
+    .. code-block:: bash 
+
+      srun -n1 /usr/bin/python3 buildbot/compile.py -o buildrocm5.7.1
+
+#. Test the compiler
+
+  Follow the steps in the next section "Using the compiler" to test this installation
+
+#. Install
+
+  #. The build step will install the compiler to the folder ``buildrocm<version>/install``. Simply copy this folder to the ``/usr/workspace/raja-dev/`` directory using the naming scheme ``clang_sycl_<git sha>_hip_gcc<version>_rocm<version>``
+
+  #. Set the permissions of the folder, and everything in it to 750
+
+    .. code-block:: bash 
```

Comment:
```suggestion
```

For simple code-like blocks, I think this syntax is simpler and easier to get formatting correct.  Same with the item below. It looks like there is something wrong with the list formatting/numbering as it is.